lib/hawkular/base_client.rb

```diff
@@ -174,14 +174,16 @@ def tenant_header
     end
 
     def handle_fault(f)
+      http_code = (f.respond_to?(:http_code) ? f.http_code : 0)
+      fail HawkularException.new('Invalid credentials', http_code) if f.instance_of? RestClient::Unauthorized
```

Comment:
I meant, try to parse the body as JSON but of course with an HTML will fail, in that case, I can return a fixed message or try to parse HTML as a regexp as you said, but honestly, I feel scary of parsing the HTML because it could change and is out of our control.

Anyway I can do the changes. :)